##Disciplina - Introdução a Ciência de Dados - Univesp
##Exercícios de Apoio da Semana 5
##**GABARITO**

###1 – Indique 3 algoritmos da biblioteca Scikit-Learn que podem ser utilizados tanto para regressão quanto para classificação.

### SVM (https://scikit-learn.org/stable/modules/svm.html)
### Stochastic Gradient Descent (SGD) (https://scikit-learn.org/stable/modules/sgd.html)
###Decision Tree (https://scikit-learn.org/stable/modules/tree.html)

### Entre outros!

###2 – Importe os dados de um arquivo CSV de avaliações de um modelo de Iphone, disponível no Kaggle (https://www.kaggle.com/kmldas/apple-iphone-se-reviews-ratings).  Uso de bibliotecas é por sua conta!

In [1]:
import pandas as pd

In [27]:
dados=pd.read_csv("APPLE_iPhone_SE.csv")

In [11]:
dados.head(5)

Ratings         Comment                                            Reviews
0        5          Super!  Great camera for pics and videos Battery life ...
1        5       Must buy!  Great device. Let me tell the Pros..1. Superb ...
2        5   Great product  Who all loves older size i.e., 4.7 inch type s...
3        5  Simply awesome  This iPhone SE is the best phone ever you get....
4        5  Classy product  This is my second iphone after iphone 4s. I’ve...

###3 – Quantas linhas tem o arquivo?

In [12]:
dados

Ratings  ...                                            Reviews
0           5  ...  Great camera for pics and videos Battery life ...
1           5  ...  Great device. Let me tell the Pros..1. Superb ...
2           5  ...  Who all loves older size i.e., 4.7 inch type s...
3           5  ...  This iPhone SE is the best phone ever you get....
4           5  ...  This is my second iphone after iphone 4s. I’ve...
...       ...  ...                                                ...
9708        5  ...                      Absolutely brilliantREAD MORE
9709        5  ...  Superb phone. This is my 4th iPhone, I feel SE...
9710        5  ...                                 very niceREAD MORE
9711        5  ...       Loving it as of now. Good Product .READ MORE
9712        5  ...                      NiceElegant ElectricREAD MORE

[9713 rows x 3 columns]

### 9713 linhas!

###4 – Qual a média das notas atribuídas?

In [5]:
dados.describe(include= 'all')

Ratings    Comment        Reviews
count   9713.000000       9713           9713
unique          NaN        264           8229
top             NaN  Wonderful  GoodREAD MORE
freq            NaN        544            245
mean       4.456399        NaN            NaN
std        1.032911        NaN            NaN
min        1.000000        NaN            NaN
25%        4.000000        NaN            NaN
50%        5.000000        NaN            NaN
75%        5.000000        NaN            NaN
max        5.000000        NaN            NaN

### Mean (média) das notas é 4.456399

###5 – Qual é a nota mais atribuída? Quantas vezes ela foi atribuída? Quantos % das notas totais?

In [9]:
!pip install sweetviz
import sweetviz

     |████████████████████████████████| 15.1 MB 187 kB/s 


In [10]:
eda=sweetviz.analyze(dados)
eda.show_html()

                                             |          | [  0%]   00:00 -> (? left)

Report SWEETVIZ_REPORT.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


In [13]:
dados['Ratings'].value_counts()

5    6788
4    1717
3     535
1     474
2     199
Name: Ratings, dtype: int64

###Usando o sweetviz eu poderia descobrir (e foi o que fiz), desta forma verifiquei em 6788 notas 5 foram atribuidas, sendo a mais atribuída. Ela representa 70% das notas.

### Também poderia usar a função value_counts() do dataframe para identificar os valores. Mas teria que montar uma estragégia para ver os valores em porcentagem

###6 – Qual a nota menos atribuída? Qual a porcentagem que ela representa no todo?

### Idem para o anterior, a nota menos atribuída é a nota 2, sendo que ela representa 2,04% das notas.

Veja que usando a contagem de notas e dividindo pelo total de notas (e multiplicando por 100) achamos a porcentagem das notas

In [14]:
dados['Ratings'].value_counts()

5    6788
4    1717
3     535
1     474
2     199
Name: Ratings, dtype: int64

In [17]:
dados['Ratings'].value_counts()/9713*100

5    69.885720
4    17.677340
3     5.508082
1     4.880058
2     2.048801
Name: Ratings, dtype: float64

###7 – Um dos comentários atribuídos é “Worth every penny” ou seja, “Vale cada centavo”. Faça a alteração nessas linhas do texto em inglês para o texto em português.

In [19]:
dados['Comment'].replace("Worth every penny","Vale cada centavo",inplace=True)

In [21]:
#só monstrando o que foi feito
dados[dados['Comment']=="Vale cada centavo"]

Ratings  ...                                            Reviews
56          5  ...  Best model if you wish to enter the Apple ecos...
115         5  ...  It's budget phone overall is the best only ban...
123         5  ...  Best phone i have ever purchased..tho battery ...
140         5  ...  Nice apple product. It works very fast and bes...
161         5  ...  Thumbs up to this new product from apple,just ...
...       ...  ...                                                ...
9609        5  ...  Overall good product but Some apps crashing on...
9611        5  ...  Except battery everything is perfect and wonde...
9643        5  ...  The phone looks awesome and is complete value ...
9659        5  ...  Phone with good performance and camera. Seamle...
9672        5  ...                        Good mobile in iOSREAD MORE

[372 rows x 3 columns]

###8 – Exclua todas as linhas em que o Comentário é “Hated it!”. Esse comentário aparece 23 vezes na lista de comentários.

In [28]:
dados.shape[0]

9713

In [29]:

dados.drop(dados.loc[dados['Comment']=='Hated it!'].index, inplace=True)


In [30]:
dados.shape[0]

9690